### PR TITLE
Tranformations parameters for pomi

### PR DESCRIPTION
**Pull request by paologallinaharbur**
This PR adds:
 - support for `transformations` parameter to be passed to nri-prometheus
 - refactor of the main (basically improved namings and extracted some functions)

Running the branch as soon as 'promhtt_' is added to the filters, matching metrics are no longer reported
Screen Shot 2021-07-09 at 6 24 01 PM

### CODE REVIEW

File: exporters/powerdns/powerdns-config.yml.sample [hunk 1]
```diff
@@ -9,3 +9,12 @@ integrations:
 
         # API key used to connect to the PowerDNS server
         # api_key: ****-***-*****
+
+        transformations:
```
Comment (gsanchezgavier) [inline exporters/powerdns/powerdns-config.yml.sample:13]: As requested by @mangulonr  this metrics should be sent by default so we could add this configuration in a form of a comment with some explanatory message.

File: nri-config-generator/args/args.go [hunk 2]
```diff
@@ -26,19 +25,15 @@ type ArgumentList struct {
 	ShortRunning bool   `default:"false" help:"By default execution is long running, but this can be override"`
 }
 
-func argVars(c *ArgumentList) (map[string]interface{}, error) {
-	err := args.SetupArgs(c)
```
Comment (gsanchezgavier) [inline nri-config-generator/args/args.go:30]: isn't it this converting from env variables to args? 
I think this helps to set configurations like verbose

File: nri-config-generator/main.go [hunk 3]
```diff
@@ -39,44 +44,68 @@ var (
 )
 
 func main() {
-	vars := map[string]interface{}{
-		varIntegrationName:    integration,
-		varIntegrationVersion: integrationVersion,
+
+	al := &args.ArgumentList{}
+	err := sdkv4.SetupArgs(al)
```
Comment (gsanchezgavier) [inline nri-config-generator/main.go:49]: oh ok i think you could discard my previous comment

File: nri-config-generator/main.go [hunk 4]
```diff
@@ -139,3 +223,17 @@ func loadConfigTemplate() (*template.Template, error) {
 	}
 	return t, nil
 }
+
+// ProcessingRule is subset of the rules supported by nri-prometheus.
+type ProcessingRule struct {
```
Comment (gsanchezgavier) [inline nri-config-generator/main.go:228]: wouldn't the args package be a better fit for this types?

File: exporters/powerdns/powerdns-config.yml.sample (hunk 1 above)
Comment (paologallinaharbur) [reply exporters/powerdns/powerdns-config.yml.sample:13]: make sense, I did not know it

File: nri-config-generator/main.go (hunk 4 above)
Comment (paologallinaharbur) [reply nri-config-generator/main.go]: makes sense, I'll change the args package name to `config`

**Review by gsanchezgavier — APPROVED** [review]
LGTM thanks!

**Pull request merged**